hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java

```diff
@@ -1532,17 +1533,18 @@ public SelectClause visitSelectClause(SqmSelectClause selectClause) {
 
 	@Override
 	public Void visitSelection(SqmSelection<?> sqmSelection) {
-		final Map<String, DomainResultProducer<?>> resultProducers;
+		final List<Pair<String, DomainResultProducer>> aliasAndProducerPairs;
```

Comment:
We might well create a simple class to combine both together. But I often feel the need of a Pair util class here and there and with good variable naming we won't impact code readability.
Let me know your preference and I am ready to revert back the Pair util class and either:

1. create a simple POJO class to combine both alias and producer;
2. use two Lists to contain alias and producer respectively (but code reader needs to keep that in mind that the sync up between them).